aten/src/ATen/Declarations.cwrap

```diff
@@ -3935,52 +3921,6 @@
         - THTensor* self
 ]]
 
-[[
-  name: select
-  cpu_half: True
-  variants: [method,function]
-  return: argument 0
-  arguments:
-    - arg: THTensor* result
-      output: True
-    - THTensor* self
-    - arg: int64_t dim
-      wrap_dim: self
-    - int64_t sliceIndex
-  aten_custom_call: |
-    int64_t ndim = self.dim();
-    AT_ASSERT(ndim > 0, "select() cannot be applied to a 0-dim tensor.");
-    if(ndim == 1) {
-      ${THTensor}_narrow(${state,}result_->tensor, self_->tensor, dim, sliceIndex,1);
-      result_->setScalar(true);
-    } else {
-      ${THTensor}_select(${state,}result_->tensor, self_->tensor, dim, sliceIndex);
-    }
-]]
-
-[[
-  name: _unnarrow
-  variants: [method,function]
-  return: argument 0
-  arguments:
-    - arg: THTensor* result
-      output: True
-    - THTensor* self
-    - arg: int64_t dimension
-      wrap_dim: self
-    - int64_t offset
-    - int64_t dimSize
-  aten_custom_call: |
-    int64_t ndim = self.dim();
```

Comment:
Are these error messages maintained in the new code?

Reply:
They're removed. We're moving towards allowing these ops on zero-dim tensors as if they were one-dim. So if `s` is a zero-dim tensor:

`s.select(dim=0, index=0)` -> `s`
`s.select(dim=0, index=1`) -> error

Zero-dim tensors aren't not exposed in Python yet so we'll have to be careful when we expose them that the backwards works properly.

Reply:
Actually, I'm not so sure the code is correct for zero-dim or if we really want to support these functions on zero-dim tensors. I'll add back the assertions.